CRM/Core/Error.php

```diff
@@ -629,19 +648,33 @@ public static function debug_query_result($query) {
   /**
    * Obtain a reference to the error log.
    *
-   * @param string $comp
+   * @param string $prefix
+   * @param array $logConfig
+   *   Configuration options to pass to the logging class.
    *
    * @return Log
    */
-  public static function createDebugLogger($comp = '') {
-    if (!isset(\Civi::$statics[__CLASS__]['logger_file' . $comp])) {
+  public static function createDebugLogger($prefix = '', $logConfig = array()) {
+    self::generateLogFileName($prefix);
+    return Log::singleton('file', \Civi::$statics[__CLASS__]['logger_file' . $prefix], '', $logConfig);
+  }
+
+  /**
+   * Get the name of the logfile to use.
+   *
+   * This function includes poor man's log file management and a check as to whether the file exists.
+   *
+   * @param string $prefix
+   *
+   * @return string
+   */
+  protected static function generateLogFileName($prefix) {
+    if (!isset(\Civi::$statics[__CLASS__]['logger_file' . $prefix])) {
       $config = CRM_Core_Config::singleton();
 
-      if ($comp) {
-        $comp = $comp . '.';
```

Comment:
This is the indentified error  - changing the value of $comp (now renamed $prefix) at this point meant it never hit the cached value